trailpanion/src/data/mapsSlice.js

```diff
@@ -0,0 +1,72 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit'
+import axios from "axios"
+import { OPEN_WEATHER_KEY } from '../constants'
+
+const initialState = {
+  trailsList: [],
+  selectedTrailIndex: -1,
+  coordinates: null
+}
+
+
+export const getCoordinates = createAsyncThunk('maps/getCoordinates', async ({city, state}) => {
```

Comment:
An action called `maps/getCoordinates` that calls the weather API to return the coordinates. See `builder.addCase(getCoordinates.fulfilled` too see what we do with the returned data